src/events/SDL_mouse.c

```diff
@@ -661,17 +661,47 @@ SDL_SendMouseWheel(SDL_Window * window, SDL_MouseID mouseID, float x, float y, S
     return posted;
 }
 
+static SDL_MouseWheelSource latest_source = SDL_MOUSEWHEEL_SOURCE_LAST;
+
+static void
+SDL_SetPanSource(SDL_MouseWheelSource source) {
+    latest_source = source;
+}
+
+static SDL_MouseWheelSource
+SDL_GetPanLastSource() {
+    return latest_source;
+}
+
+int
+SDL_SendPanDelta(
+        SDL_Window * window,
+        SDL_MouseID mouseID,
+        double delta,
+        SDL_PanAxis axis
+) {
+    SDL_Mouse *mouse = SDL_GetMouse();
+
+    if( SDL_GetEventState(SDL_PANEVENT) == SDL_ENABLE ) {
+        SDL_Event event;
+        event.type = SDL_PANEVENT;
```

Comment:
Is timestamp missing ?

Reply:
Timestamp is added automatically by event subsystem